src/Mono.Android/Android.Runtime/AndroidEnvironment.cs

```diff
@@ -426,11 +427,7 @@ public Uri GetProxy (Uri destination)
 				if (address == null) // FIXME
 					return destination;
 
-#if ANDROID_19
-				return new Uri (string.Format ("http://{0}:{1}/", address.HostString, address.Port));
-#else
-				return new Uri (string.Format ("http://{0}:{1}/", address.HostName, address.Port));
-#endif
+				return new Uri (FormattableString.Invariant ($"http://{address.HostName}:{address.Port}/"));
```

Comment:
I think this logic might be "backwards": API-19+ should be using `address.HostString`, while Android. up through API-18 should use `address.HostName`.

Verifying this, [`InetSocketAddress.getHostString()`](https://developer.android.com/reference/java/net/InetSocketAddress#getHostString()) was added in API-19, while [`InetSocketAddress.getHostName()`](https://developer.android.com/reference/java/net/InetSocketAddress#getHostName()) was added in *API-1*.

This should be using `address.HostString`.